### PR TITLE
allow tmpfs.size to be set in unit bytes

### PR DESCRIPTION
**Pull request by ndeloof**
see https://github.com/compose-spec/compose-spec/pull/205

Close https://github.com/docker/compose/issues/8696

### CODE REVIEW

File: schema/compose-spec.json [hunk 1]
```diff
@@ -426,8 +427,10 @@
                     "type": "object",
                     "properties": {
                       "size": {
-                        "type": "integer",
-                        "minimum": 0
+                        "oneOf": [
+                          {"type": "integer","minimum": 0},
```
Comment (mat007) [inline schema/compose-spec.json:431]: nit:
```suggestion
                          {"type": "integer", "minimum": 0},
```

File: schema/compose-spec.json [hunk 2]
```diff
@@ -599,12 +602,12 @@
...
-          "device_ids": {"$ref": "#/definitions/list_of_strings"},
-          "driver":{"type": "string"},
-          "options":{"$ref": "#/definitions/list_or_dict"}
-        },
+            "capabilities": {"$ref": "#/definitions/list_of_strings"},
+            "count": {"type": ["string", "integer"]},
+            "device_ids": {"$ref": "#/definitions/list_of_strings"},
+            "driver":{"type": "string"},
+            "options":{"$ref": "#/definitions/list_or_dict"}
+          },
```
Comment (mat007) [inline schema/compose-spec.json:610]: Indentation seems to be 2 characters in the rest of the file, was this change intended?

File: loader/full-struct_test.go [hunk 3]
```diff
@@ -866,7 +866,7 @@ func fullExampleYAML(workingDir, homeDir string) string {
     - type: tmpfs
       target: /opt
       tmpfs:
-        size: 10000
+        size: "10000"
```
Comment (mat007) [inline loader/full-struct_test.go:869]: Is this change intentional?

File: loader/loader_test.go [hunk 4]
```diff
@@ -936,15 +936,15 @@ services:
       - type: tmpfs
         target: /app
         tmpfs:
-          size: 10000
+          size: 1kb
```
Comment (mat007) [inline loader/loader_test.go:939]: Do we want a test with both allowed formats? (or two tests)

File: loader/full-struct_test.go (hunk 3 above)
Comment (ndeloof) [reply loader/full-struct_test.go:869]: yes, using UnitBytes makes the marshaller always prefer string

File: schema/compose-spec.json (hunk 2 above)
Comment (ndeloof) [reply schema/compose-spec.json:610]: I copied the exact content from compose-spec - we haven't yet found a way to directly refer to the reference repository for this file

Comment (ulyssessouza) [reply schema/compose-spec.json:610]: Maybe a JSON linter and the same approach as we use on `V1`, which is to integrate the download of the spec file in the build.

**Review by ulyssessouza — APPROVED** [review]
LGTM
